app/assets/javascripts/host_checkbox.js

```diff
@@ -86,14 +91,40 @@ function cleanHostsSelection() {
   return false;
 }
 
+function multiple_selection(){
+  var count = $("#pagination").data("meta-count");
+  var alert_text = Jed.sprintf(__("All") + "<b> %d </b>" + __("hosts are selected. "), count);
+  var undo_text = __("Undo selection");
+  $(".text").html('<span id="multiple-chosen">' + alert_text + '<a href="#" onclick="undo_multiple_selection();">' + undo_text + '</a></span>');
+  $("#multiple-chosen").data('multiple', true)
+}
+
+function undo_multiple_selection(){
+  var count = $("#pagination").data("meta-count");
+  var per_page = $("meta[name='itemsPerPage']").attr("content");
+  var alert_text = Jed.sprintf(__("All %s hosts on this page are selected. "), per_page);
+  var select_text = Jed.sprintf(__("Select all") + "<b> %s </b>" + __("hosts"), count);
```

Comment:
This won't translate correctly in all languages as the strings `Select all` and `hosts` will be translated separately and not as one sentence. Also above on line 96.